### PR TITLE
authentication activated with next auth

### PR DESCRIPTION
**Pull request by Waislam**
team page design’ and layout added to member detail page
added all types of modal to team page. Authentication is working now with next auth. Still has some issues which will be done in the next pull. Right now the login auth is working.

### CODE REVIEW

File: frontend/pages/dashboard.tsx [hunk 1]
```diff
@@ -1,9 +1,21 @@
 import { LoanIcon, LoanPlusIcon, PersonPlusIcon } from '@/icons';
 import DashboardLayout from '@/Layouts/DashboardLayout';
 import Analytics from '@/modules/dashboard/components/Analytics';
-import { ReactNode } from 'react';
+import { ReactNode, useEffect } from 'react';
+import { useSession, signIn, signOut } from 'next-auth/react';
+import { useRouter } from 'next/router';
+
 
 const Dashboard = () => {
```
Comment (tusheer) [inline frontend/pages/dashboard.tsx:9]: I think this redirection should be done using the getServerSideProps

image

**Review by tusheer — CHANGES_REQUESTED** [review]

File: frontend/pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import NextAuth from "next-auth/next";
+import CredentialsProvider from "next-auth/providers/credentials";
+import api from "../api";
+import React from "react";
+import { type } from "os";
+
+const backendURL = process.env.NEXT_PUBLIC_BACKEND_URL;
+
+type UserCredential{
```
Comment (fazlay) [inline frontend/pages/api/auth/[...nextauth].tsx:9]: type UserCredential = {
    username: string;
    password: string;
};

File: frontend/pages/api/auth/[...nextauth].tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    username: string,
+    password: string,
+}
```
Comment (fazlay) [inline frontend/pages/api/auth/[...nextauth].tsx:12]:
```
type UserCredential = {
    username: string;
    password: string;
};
```

File: frontend/pages/login.tsx [hunk 4]
```diff
@@ -64,4 +87,14 @@ Login.getLayout = (page: ReactNode) => {
     return <BaseLayout className="flex min-h-screen flex-col bg-brand-25/50">{page}</BaseLayout>;
 };
 
+export async function getServerSideProps(context) {
```
Comment (fazlay) [inline frontend/pages/login.tsx:90]:
```suggestion
export async function getServerSideProps(context:GetServerSidePropsContext) {
```

**Pull request closed without merging**